src/libs/error.js

```diff
@@ -63,3 +63,11 @@ export const withErrorReportingInModal = _.curry((title, onDismiss, fn) => async
 export const withErrorReporting = _.curry((title, fn) => {
   return withErrorIgnoring(reportErrorAndRethrow(title)(fn))
 })
+
+
+/**
+ * Invoke the `callback` with any error thrown when evaluating the async `fn` with `...args`.
+ */
+export const onError = _.curry((callback, fn) => async (...args) => {
```

Comment:
Many (most?) functions in Terra UI that wrap other functions follow the naming pattern `with...`. Maybe something like `withErrorHandling`?

Also, it looks like most of the other `with...` functions in this module could be implemented using this.

Reply:
Would you mind taking a look at this again and making sure you're happy with the new implementations?

Reply:
ok they're wrong. I'll fix them.

Reply:
Fixed - can you take a look?